### PR TITLE
[Table] Add emphasized borders for frozen columns and rows

### PR DESCRIPTION
**Pull request by michael-yx-wu**
#### Changes proposed in this pull request:

Add 2px inset box-shadow to frozen columns and rows.

screen shot 2017-10-10 at 7 10 46 pm

screen shot 2017-10-10 at 7 11 29 pm

#### Reviewers should focus on

- Colors? I picked blue4 for light theme and blue1 for dark theme
- I refactored the border styles to be more readable (would be nice if everything could just be inset styles in a future PR)

### CODE REVIEW

File: packages/table/src/table.tsx [hunk 1]
```diff
@@ -1157,9 +1157,9 @@ export class Table extends AbstractComponent<ITableProps, ITableState> {
         const columnIndices = this.grid.getColumnIndicesInRect(viewportRect, fillBodyWithGhostCells);
 
         const columnIndexStart = showFrozenColumnsOnly ? 0 : columnIndices.columnIndexStart;
-        const columnIndexEnd = showFrozenColumnsOnly ? numFrozenColumns : columnIndices.columnIndexEnd;
+        const columnIndexEnd = showFrozenColumnsOnly ? numFrozenColumns - 1 : columnIndices.columnIndexEnd;
```
Comment (michael-yx-wu) [inline packages/table/src/table.tsx]: this (and the row below) are fencepost errors. `tableBodyCell` actually renders up to and including the end indices: https://github.com/palantir/blueprint/blob/783d8e26fd11c5a6d387d48152defe687865c74f/packages/table/src/tableBodyCells.tsx#L127

**Comment by llorca** [comment]
Nice stuff, but def not blue my friend. Keep the 2px width and reuse the same color we use for every border

**Comment by cmslewis** [comment]
Visual bugs on hover:
- [ ] image
- [ ] image

Visual bugs (extra borders) in `MAIN` quadrant:
- [ ] image
- [ ] image

File: packages/table/src/table.tsx (hunk 1 above)
Comment (cmslewis) [reply packages/table/src/table.tsx]: You can remove changes in this file. These fencepost errors are already fixed in #1678.

**Comment by michael-yx-wu** [comment]
Changed colors to the default border color. Still 2px. Fixed hover bugs and removed freeze border styling from cells that didn't need it.

File: packages/table/src/cell/_common.scss [hunk 2]
```diff
@@ -13,9 +13,11 @@ $large-cell-height: $pt-grid-size * 3 !default;
 $cell-background-color: $white !default;
 $cell-background-color-opacity: 0.1 !default;
 $cell-border-color: $pt-divider-black !default;
+$cell-border-color-frozen: $blue4;
```
Comment (cmslewis) [inline packages/table/src/cell/_common.scss]: I think you can delete these now.

File: packages/table/src/cell/_common.scss [hunk 3]
```diff
@@ -13,9 +13,11 @@ $large-cell-height: $pt-grid-size * 3 !default;
...
 $cell-text-color: $pt-text-color !default;
 $dark-cell-background-color: $dark-gray3 !default;
 $dark-cell-border-color: $pt-dark-divider-black !default;
+$dark-cell-border-color-frozen: $blue1;
```
Comment (cmslewis) [inline packages/table/src/cell/_common.scss]: Same

**Comment by giladgray** [comment]
the gray borders look a little too light, not obviously intentional as a separator. maybe one shade darker or 3px width?

**Comment by cmslewis** [comment]
+1 to @giladgray's thoughts, though I'd vote for a darker color over a thicker 3px border. Infringing on the cell content by an additional pixel may start to look imbalanced.

**Comment by llorca** [comment]
I'd actually prefer trying 3px width (hey, it matches resize handle and reorder indicator) -- borders of different color is visually gross

**Comment by michael-yx-wu** [comment]
Border of different colors:
screen shot 2017-10-12 at 7 35 42 am
screen shot 2017-10-12 at 7 35 36 am

**Comment by michael-yx-wu** [comment]
Here's what it looks like at 3px but same color:

screen shot 2017-10-12 at 9 48 03 am
screen shot 2017-10-12 at 9 48 07 am

**Comment by michael-yx-wu** [comment]
Ah yeah I've noticed the overlap but I've ignored it since it's barely noticeable at 1px. At 3px you really can tell :P

**Comment by cmslewis** [comment]
Thanks @michael-yx-wu!

**Pull request merged**